block/manager.go

```diff
@@ -934,19 +936,22 @@ func (m *Manager) submitBlocksToDA(ctx context.Context) error {
 				gasPrice = gasPrice * m.dalc.GasMultiplier
 			}
 			m.logger.Info("retrying DA layer submission with", "backoff", backoff, "gasPrice", gasPrice, "maxBlobSize", maxBlobSize)
-			time.Sleep(backoff)
-			backoff = m.exponentialBackoff(backoff)
+
 		case da.StatusTooBig:
-			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
 			maxBlobSize = maxBlobSize / 4
-			time.Sleep(backoff)
-			backoff = m.exponentialBackoff(backoff)
+			fallthrough
 		default:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
-			time.Sleep(backoff)
 			backoff = m.exponentialBackoff(backoff)
 		}
+
 		attempt += 1
+
+		select {
+		case <-ctx.Done():
+			break daSubmitRetryLoop
+		case <-time.After(backoff):
+		}
```

Comment:
Alright. Last change :-) 

Now that we set the `backoff` to zero initially, we can move this select statement to the beginning of the for loop. The first iteration will not block because `<-time.After(backoff)` will immediately execute since `backoff == 0`. This will avoid us sleeping on the last iteration where `attempts == mazSubmitAttempts`.

Reply:
that...  makes so much sense @_@, didn't think of it

updated it!